src/main.cpp

```diff
@@ -2832,7 +2847,22 @@ bool LoadExternalBlockFile(FILE* fileIn)
                     blkdat >> block;
                     if (ProcessBlock(nullptr, &block, false)) {
                         nLoaded++;
-                        LogPrintf("Blocks/s: %f", nLoaded / ((GetTimeMillis() - nStart) / 1000.0));
+                        if (display_progress) {
+                            unsigned int percent_progress = percent_start + (uint64_t) nPos
+                                    * (uint64_t) (percent_end - percent_start) / file_size;
+
+                            LogPrintf("INFO: %s: blocks/s: %f, progress: %u%%", __func__,
+                                      nLoaded / ((GetTimeMillis() - nStart) / 1000.0), percent_progress);
```

Comment:
Might make sense to rate limit this a bit, together with `SetBestChain` log messages this creates quite a bit of spam.

Reply:
I actually agree. This is original code. In the code I added, I rate limited the messages to only when the % actually changes (at the integer level), which means only 100 messages max for the whole thing. Note that the file size parameters are optional, in which case something else needs to be used, maybe mod 1000 like a few other areas. I also hate the block verification progress messages (in a slightly different area), because they are misleading. The block verification is actually done in reverse from the head downwards to 1000 blocks depth by default.